src/device.rs

```diff
@@ -120,8 +117,11 @@ impl Device {
     }
 
     // Maybe improve this API?
-    pub fn acquire(&self) -> Status {
-        unsafe { Status::from(ffi::cairo_device_acquire(self.to_raw_none())) }
+    pub fn acquire(&self) -> Result<(), Error> {
```

Comment:
Indeed, let's return a value here that calls `release()` on `Drop`. The way it is now is asking for problems

Reply:
hmm, I'm not sure how to do that. I just wanted to remove the Status usage here. What do you suggest to use?

Reply:
```rust
pub struct DeviceAcquireGuard<'a>(&'a Device);

impl<'a> Drop for DeviceAcquireGuard<'a> {
    fn drop(&mut self) { self.0.release(); }
}
```

And return that from `acquire()`

Reply:
(and derive `Debug`)

Reply:
(and make `release()` private)

Reply:
Done, thanks for the pointer :)